packages/fetchai/connections/http_server/connection.py

```diff
@@ -533,6 +533,7 @@ async def connect(self) -> None:
         """
         if self.is_connected:
             return
+
         self._state.set(ConnectionStates.connecting)
         self.channel.logger = self.logger
         await self.channel.connect(loop=self.loop)
```

Comment:
Regarding the loop parameter; I guess this and other connection channels could be changed accordingly (just use `asyncio` loop getter)? Not crucial though; it could be done in another moment.

Reply:
supposed that channel is internal implementation and used by the connection, channel have to share the same loop as connection uses, so would stay explicit loop set for channel. and if we will change the loop parameter of connection it will be easier to change just loop property of connection instead of changing loop property of connection and channel